src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -227,6 +227,22 @@ private function parseDefaults(array &$content, $file)
     }
 
     /**
+     * @param array $service
+     *
+     * @return bool
+     */
+    private function isUsingShortSyntax(array $service)
+    {
+        foreach ($service as $key => $value) {
+            if (!is_int($key) && '$' !== substr($key, 0, 1)) {
```

Comment:
should be an `||` instead of `&&`?
I'd write it as `if (!isset($key[0]) || '$' !== $key[0]) {`